### ISSUE TITLE
Phantom/duplicate notification causing vibration

### ISSUE THREAD

**Issue by colleenm** [signalapp/Signal-Android#9306]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Signal sometimes causes my phone to vibrate when I have an old notification open that I have not yet cleared.

### Steps to reproduce
- Receive message in a Signal conversation
- Do not clear notification from system notification bar
- Wait some indeterminate and seemingly random period of time without receiving any new messages in that conversation

**Actual result:** Phone vibrates. The Android system notification log shows the vibration was caused by the Signal conversation with a notification in the system status bar
**Expected result:** Nothing, because there aren't new messages in the conversation

### Screenshots
(can't show useful screenshots without PII)

### Device info

**Device:** Pixel 2
**Android version:** 8.1.0
**Signal version:** 4.54.4

### Link to debug log
https://debuglogs.org/0d86bca0bee0f6fa46dc5c0b26787e9d98d70ffdec2c23067c24047696954a08

The buggy notification behavior happened at 16:32, according to the Android system notification log.

Relevant notifications settings:
Notifications: on
Sound: Default
Vibrate: on
Repeat alerts: Never
Show: Name and message
Priority (system setting): "High: Make sound"

**Comment by oittaa** [comment]
Same issue with Pixel 3 XL, Android 10, Signal 4.53.7.

**Comment by oittaa** [comment]
Still happening with the latest version. Some people in Reddit have the same issue. https://www.reddit.com/r/signal/comments/f6f5zh/duplicate_notifications/

**Comment by SwartzCr** [comment]
I too have been having this issue EXCEPT that I think I know what's causing it!

If you have a conversation that is muted (such as a high traffic group) and an unread notification that you have dismissed from your homescreen (but not marked as read) - when the muted group receives a notification that would normally be muted it will cause the unread notification to re-appear (and re trigger a notification vibration).

Hopefully this will help in reproducing and debugging this!

**Comment by cascheberg** [comment]
I also experienced this multiple times a while ago (in February; long ago because usually I have my phone set to DnD). For one case I have links to debug logs from my phone and from the contact I was writing to, but the content is not accessible anymore.

Some facts that I had written down from the logs:
At 13:06:27 I received a message for "_thread 4_" (in logs) which posts a notification. So this is when I received the actual message and saw the notification, I did not dismiss the notification. The sender sent the message at 13:06:06.
Shortly after at 13:06:31 again a notification is posted for "_thread 4_". In my logs I see "_Typing stopped on thread 4 due to an incoming message_”. In the sender's debug log I see that at 13:06:31 he received a delivery receipt.

I have a Signal Desktop instance linked to my phone, but it was not running. I have typing indicators disabled, read receipts enabled. I only know that the sender has read receipts enabled as well.
